crates/runtime/src/memory.rs

```diff
@@ -137,22 +137,23 @@ impl RuntimeLinearMemory for MmapMemory {
         self.maximum
     }
 
-    fn grow_to(&mut self, new_size: usize) -> Option<()> {
+    fn grow_to(&mut self, new_size: usize) -> Result<()> {
         if new_size > self.mmap.len() - self.offset_guard_size - self.pre_guard_size {
             // If the new size of this heap exceeds the current size of the
             // allocation we have, then this must be a dynamic heap. Use
             // `new_size` to calculate a new size of an allocation, allocate it,
             // and then copy over the memory from before.
             let request_bytes = self
                 .pre_guard_size
-                .checked_add(new_size)?
-                .checked_add(self.extra_to_reserve_on_growth)?
-                .checked_add(self.offset_guard_size)?;
+                .checked_add(new_size)
+                .ok_or_else(|| format_err!("overflow: new size"))?
+                .checked_add(self.extra_to_reserve_on_growth)
+                .ok_or_else(|| format_err!("overflow: extra to reserve on growth"))?
+                .checked_add(self.offset_guard_size)
+                .ok_or_else(|| format_err!("overflow: offset guard size"))?;
```

Comment:
I think that having a different error-per-overflow may be a bit confusing here, but an error message for any overflow all seems reasonable

Reply:
Agreed